libraries/freertos_plus/standard/freertos_plus_posix/source/FreeRTOS_POSIX_utils.c

```diff
@@ -186,34 +186,55 @@ int UTILS_TimespecAdd( const struct timespec * const x,
                        const struct timespec * const y,
                        struct timespec * const pxResult )
 {
-    int64_t llPartialSec = 0;
+    uint64_t llPartialSec = 0;
     int iStatus = 0;
-
-    /* Check parameters. */
-    if( ( pxResult == NULL ) || ( x == NULL ) || ( y == NULL ) )
+    struct timespec temp = { .tv_sec = 1, .tv_nsec = 1 };
+    uint8_t isSigned = 1u;
+
+    /* Check input parameters. None of the parameters should be NULL,
+     * and the input parameters values should be positive. */
+    if( ( pxResult == NULL ) || ( x == NULL ) || ( y == NULL ) ||
+        ( x->tv_sec < 0 ) || ( x->tv_nsec < 0 ) ||
+        ( y->tv_sec < 0 ) || ( y->tv_nsec < 0 ) )
     {
         iStatus = -1;
     }
 
+    configPRINTF( ( "Input tv_sec values: x->tv_sec: %u, y->tv_sec:%u",
+                    x->tv_sec, y->tv_sec ) );
+
     if( iStatus == 0 )
     {
-        /* Perform addition. */
+        /* Perform addition for nanoseconds result. */
         pxResult->tv_nsec = x->tv_nsec + y->tv_nsec;
 
-        /* check for overflow in case nsec value was invalid */
-        if( pxResult->tv_nsec < 0 )
+        configPRINTF( ( "Values after nsec addition: tv_nsec=%ld, tv_sec=%d\n",
+                        pxResult->tv_nsec, pxResult->tv_sec ) );
+
+        /* Check if the addition resulted in an overflow.*/
+        if( ( pxResult->tv_nsec < x->tv_nsec ) || ( pxResult->tv_nsec < y->tv_nsec ) )
         {
+            configPRINTF( ( "Detected overflow in tv_nsec: ReturnStatus=%d", iStatus ) );
```

Comment:
This will print incorrect ReturnStatus. `iStatus` needs to be sets first before printing.

Reply:
These log statements are only for debugging. Will remove them once I have a successful test execution result for Microchip on CI